src/bsoncxx/view_or_value.hpp

```diff
@@ -118,7 +118,7 @@ class view_or_value {
         return _view;
     }
 
-   private:
+   protected:
```

Comment:
I think we need access to _value in order to do c_str() the right way.  If we make a string copy off of the view, which we can access through view(), the copied string is temporary and the char* returned by c_str() ends up dangling.  I could add a public or protected getter and setter for _value and use those instead of accessing directly, would that be better?